polaris-react/src/components/Box/Box.tsx

```diff
@@ -207,24 +190,29 @@ export const Box = forwardRef<HTMLElement, BoxProps>(
     },
     ref,
   ) => {
+    // eslint-disable-next-line no-nested-ternary
+    const borderStyleValue = borderStyle
+      ? borderStyle
+      : borderColor || borderWidth
```

Comment:
Should this also be checking if `borderBlockStartWidth`, `borderBlockEndWidth`, etc. are passed in in case `borderWidth` isn't but we still want to set the `solid` border style?
```suggestion
      : borderColor ||
        borderWidth ||
        borderBlockStartWidth ||
        borderBlockEndWidth ||
        borderInlineStartWidth ||
        borderInlineEndWidth
```